kernel/src/process.rs

```diff
@@ -261,6 +261,24 @@ pub fn load_processes<C: Chip>(
     Ok(())
 }
 
+/// Opaque identifier for custom grants allocated dynamically from a process's
+/// grant region.
+///
+/// This type allows Process to provide a handle to a custom grant within a
+/// process's memory that `ProcessGrant` can use to access the custom grant
+/// memory later.
+///
+/// We use this type rather than a direct pointer so that any attempt to access
+/// can ensure the process still exists and is valid, and that the custom grant
+/// has not been freed.
+///
+/// The fields of this struct are private so only Process can create this
+/// identifier.
+#[derive(Copy, Clone)]
+pub struct ProcessCustomGrantIdentifer {
```

Comment:
In the future (especially with the reorganization of `kernel::process`) one might want to look into making this a type parameter on the Process trait, but it seems totally fine to have that type used for all processes right now.